docs/how-to-guides/test-an-app.md

```diff
@@ -8,161 +8,168 @@ keywords: testing, qa, quality assurance, test runner
 
 OPTIONS
-        --extension-dir <extensionDir>, -e <extensionDir>
-            A directory containing extensions.
+        --temp-folder <tempFolder>, -tf <tempFolder>
+            A folder to store temporary files
 
             This option may occur a maximum of 1 times
 
 
-        --input-file <inputFile>, -i <inputFile>
-            A JSON file containing the input records.
+        --test-directory <testDirectory>, -td <testDirectory>
+            A directory containing SQL files to test.
 
             This option may occur a maximum of 1 times
+```
 
+## File structure
 
-        --output-file <outputFile>, -o <outputFile>
-            A JSON file containing the expected output records.
+Here is a sample test file:
 
-            This option may occur a maximum of 1 times
+```sql
+--@test: Passing test
 
+CREATE STREAM s (id INT KEY, foo INT) WITH (kafka_topic='s', value_format='JSON');
+CREATE TABLE t (id INT PRIMARY KEY, bar INT) WITH (kafka_topic='t', value_format='JSON');
 
-        --sql-file <statementsFile>, -s <statementsFile>
-            A SQL file containing KSQL statements to be tested.
+CREATE STREAM j AS SELECT s.id, s.foo, t.bar FROM s JOIN t ON s.id = t.id;
 
-            This option may occur a maximum of 1 times
+INSERT INTO t (rowtime, id, bar) VALUES (1, 1, 1);
+INSERT INTO s (rowtime, id, foo) VALUES (1, 1, 2);
+
+ASSERT VALUES j (rowtime, s_id, foo, bar) VALUES (1, 1, 2, 1);
+
+--@test: Failing test
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: (The following columns are changed, missing or reordered: [`COL2` INTEGER])
+
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM a (id INT KEY, col1 INT, col2 INT) WITH (kafka_topic='a', value_format='JSON');
+CREATE STREAM b AS SELECT id, col1, col2 FROM a;
+
+INSERT INTO a (id, col1) VALUES (3, 5);
+
+ASSERT VALUES b (id, col1) VALUES (3, 5);
+
+CREATE OR REPLACE STREAM b AS SELECT id, col1 FROM a;
 ```
 
-## Test file structure
+A test file contains one or more tests separated by the `--@test` directive.
+Each test consists of KSQL comments containing directives and sql stataments.
+
+### Directives
 
-### Statements File
+There are three directives available:
 
-The statements file contains the SQL statements to test. The following
-are the supported statements in the testing tool:
+* `--@test: name of test`. This is required in every test.
+* `--@expected.error: error class`. This directive checks that the test throws an error of the
+provided type. 
+* `--@expected.message: error message`. This directive checks that the test throws an error with a
+message containing the provided message. 
+
+### Statements
+
+`run-ksql-test` runs each statement in a test sequentially until an error is thrown or the end of
+the test is reached.
+
+The following are the supported KSQL statements in `run-ksql-test`:
 
 - `CREATE STREAM`
 - `CREATE TABLE`
 - `CREATE STREAM AS SELECT`
 - `CREATE TABLE AS SELECT`
+- `ALTER STREAM`
```

Comment:
I'd also add a link to respective pages in the docs for every KSQL statement (could surely impact people's curiosity to learn even more, mine for sure)